### PR TITLE
Issue #ED-1613 fix: Request getting timed out for district with more than 17 character length

### PR DESCRIPTION
**Pull request by 5Amogh**
# SunbirdEd - Portal
---

### Request getting timed out for district with more than 17 character length.
---

### Type of change

Please choose appropriate options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]
No unit tests has written for hot-fix also?

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts [hunk 1]
```diff
@@ -551,7 +551,8 @@ export class DatasetsComponent implements OnInit, OnDestroy {
     this.globalDistrict = $event.value;
     this.reportForm.controls.districtName.setValue($event.value);
     this.displayFilters['District'] = [$event?.source?.triggerValue]
-    this.tag =  _.get(this.reportForm, 'controls.solution.value')+ '_' + this.userId+'_'+ _.toLower(_.trim([$event?.source?.triggerValue]," "));
+    const tagId =  _.get(this.reportForm, 'controls.solution.value')+ '_' + this.userId+'_'+ _.toLower(_.trim([$event?.source?.triggerValue]," "));
+    this.tag = tagId.length > 79 ? tagId.substring(0,79) : tagId
```
[the remark this answers is not in the document]
Comment (5Amogh) [reply src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:555]: Unit tests have been added, Please check
.
As the solution filter is a mandatory field before executing this function we already will have a solution value thus it'll not be undefined.

[the remark this answers is not in the document]
Comment (5Amogh) [reply src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:555]: The tagId to request a report allows only upto 79 characters in the URL thus we had to trim the rest of the other characters which were causing the timed out issue

[the remark this answers is not in the document]
Comment (vinukumar-vs) [reply src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:555]: But this is not a good way of trimming. Either you limit the user input field length(_.get(this.reportForm, 'controls.solution.value')) or dont add too many parameter to tagID.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (5Amogh) [reply src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:555]: The solution value is not a user entered value it's the id of the solution selected and we restrict that.
and yes this is only for short term, we are considering a long term approach using md5 hashing and migration of existing reports which have been generated using this approach.

Please check this discussion where we have asked the approaches we need to take for long term solution 
 https://github.com/orgs/Sunbird-Obsrv/discussions/62

the sample of tagId = 'xxxx9beb823d601f0af5xxxx_xxxx3686-0e88-4482-b86f-b34bbfb7xxxx_bengaluru urban s'

**Comment by vinukumar-vs** [comment]
@kirang20 @vijiurs 
As discussed, this is the temporary solution for hot-fix to unblock the on-going programs. Immediately annother hot-fix will be pushed with properly as discussed in the below thread.
https://github.com/orgs/Sunbird-Obsrv/discussions/62

@vrayulu @SanthoshVasabhaktula @Krishnaj20

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**